src/pages/lessons/[slug].tsx

```diff
@@ -49,44 +50,42 @@ const lessonQuery = /* GraphQL */ `
 
 const fetcher = (url: RequestInfo) => fetch(url).then((r) => r.json())
 
+const getNextUpData = (url: string) => {
+  const {data} = useSWR(url, fetcher)
+  return data
+}
+
 type NextUpProps = {
-  url: string
+  data: {
+    list: {
+      lessons: Types.Lesson[]
```

Comment:
Created the `Lesson` type and stored it in `types.ts` - I think we will use this type on other pages also?

Reply:
I would prefer not to have a lesson type currently or consider a more generic resource type. We’ve discussed generating these types from our GraphQL schema on the egghead side but haven’t come to a resolution

Probably fine for now, but that’s why there wasn’t a Lesson type.